application/helpers/remotecontrol/remotecontrol_handle.php

```diff
@@ -3309,7 +3310,14 @@ public function get_uploaded_files($sSessionKey, $iSurveyID, $sToken)
             return array('status' => 'No permission');
         }
 
-        $oResponses = Response::model($iSurveyID)->findAllByAttributes(array('token' => $sToken));
+        if (empty($responseId)) {
+            if (empty($sToken)) {
+                return ['status' => 'Invalid arguments: both Token and Reponse ID are empty'];
+            }
+            $oResponses = Response::model($iSurveyID)->findAllByAttributes(array('token' => $sToken));
+        } else {
+            $oResponses = [Response::model($iSurveyID)->findByPk($responseId)];
```

Comment:
My opinion, must check if token is not empty and use it in request if not empty.

Reply:
You mean Token overrules Response id?

Reply:
1. responseId and token : `token=:token AND id=:responseid`
2. only responseId `id=:responseid`
3. only token `token=:token`

Something like that : 

```
    if (empty($responseId) and empty($sToken)) {
        return ['status' => 'Invalid arguments: both Token and Reponse ID are empty'];
    }
    $criteria = new CDbCriteria();
    if (!empty($responseId)) {
        $criteria->compare('id', $responseId);
    }
    if (!empty($sToken)) {
        $criteria->compare('token', $sToken);
    }
    $oResponses = Response::model($iSurveyID)->findAll($criteria);
    if (empty($oResponses)) {
        return array('status' => 'Could not find response for given token');
    }
    // Go go !
```

No cast as array, return error more clear (in my opinion).

Reply:
> No cast as array, return error more clear (in my opinion).

What do you mean?

Reply:
> * responseId and token : `token=:token AND id=:responseid`
> * only responseId `id=:responseid`
> * only token `token=:token`

I understand. Wondering what's the best choice. Combining filters or make one overrule the other. Keep me thinking.
@olleharstedt ? What do you think?

Reply:
You guys gonna make me think again?

Reply:
You can do that on Monday. Friday is off for thinking!

Reply:
> > * responseId and token : `token=:token AND id=:responseid`
> > * only responseId `id=:responseid`
> > * only token `token=:token`
> 
> I understand. Wondering what's the best choice. 

**My** opinion : calling with token AND srid : developer think he has token AND srid, not only srid :).

Because he can always call with only srid

Reply:
You think i'm wring here : **calling with token AND srid : developer think he has token AND srid, not only srid**

?

Reply:
> I need srid 22 IF token is tok

I really doubt if this is a common scenario or if we should implement it.
Still, as it is a rare case, I think either decision will be just fine.

Reply:
@Shnoulle You wanna throw an error if both token and response id are given?

Reply:
No : i return an error if  both token and response id are set AND reponseId are not related to this token.

Remind : you can have multiple response if for one token.

Reply:
My exact solution are this one : 

```
    if (empty($responseId) and empty($sToken)) {
        return ['status' => 'Invalid arguments: both Token and Reponse ID are empty'];
    }
    $criteria = new CDbCriteria();
    if (!empty($responseId)) {
        $criteria->compare('id', $responseId);
    }
    if (!empty($sToken)) {
        $criteria->compare('token', $sToken);
    }
    $oResponses = Response::model($iSurveyID)->findAll($criteria);
    if (empty($oResponses)) {
        return array('status' => 'Could not find response for given token');
    }
    // Go go !
```

- token is set : i use it when search
- srid is set : i use it when search

Reply:
> No : i return an error if both token and response id are set AND reponseId is not related to this token.

Sounds reasonable.

> Remember: you can have multiple responses for one token.

Really? Didn't know.

Reply:
> Really? Didn't know.

Use left > 1 or "[Allow multiple response](https://manual.limesurvey.org/Participant_settings/en#Allow_multiple_responses_or_update_responses_with_one_token)" setting i use it a lot …

Reply:
@gabrieljenik Can you update, please?

Reply:
> Remind : you can have multiple response if for one token.

Right